measure

```diff
@@ -0,0 +1,181 @@
+                out_reg = m_data.get('output_regex')
+                assert isinstance(out_reg, str), 'The `output_regex` attribute is required in metric `{}` under `batch` section and must be a string, found {}'.format(m_name, type(metrics).__name__)
+                try:
+                    re.compile(out_reg)
+                except re.error as e:
+                    raise Exception('Failed to compile the `output_regex` of metric `{}` in `batch` section'.format(m_name)) from e
+
+    def describe(self):
+        metrics = { 'total_runtime': { 'unit': 'seconds' } }
+        for m_name, m_data in self.config.get('metrics', {}).items():
+            metrics[m_name] = {}
+            unit = m_data.get('unit')
+            if unit:
+                metrics[m_name]['unit'] = unit
+
+        return metrics
+
+    def _print_progress_wrapper(self):
+        if self.start_time:
+            elapsed_time = time.time() - self.start_time
+            self.progress = int(round(elapsed_time / self.config.get('expected_duration', 3600) * 100))
+        
+        super().print_progress()
+
+    def measure(self):
+        try:
+            in_metrics = self.input_data['metrics']
+        except:
+            raise Exception('Measure input is missing "metrics" key')
+
+        control = self.input_data.get('control')
+        state = state_store.get_state()
+        # Construct a state from config defaults if batch adjust has not occurred yet 
+        if not state:
+            def_state_comps = {}
+            for c_name, c_data in self.config["application"]["components"].items():
+                def_setts = {}
+                for s_name, s_data in c_data["settings"].items():
+                    def_setts[s_name] = { "value": s_data['default'] }
+                def_state_comps[c_name] = { "settings": def_setts }
+            
+            state = { "application": { "components": def_state_comps } }
+
+        self.progress_message = "Formatting command"
+        self.print_progress()
+        
+        try:
+            formatted = self.config['command'].format(state=state, control=control)
+        except Exception as e:
+            raise Exception('Failed to construct formatted command from string `{}`. \nstate: {} \n\ncontrol: {}'.format(self.config['command'], state, control)) from e
+
+
+        self.progress_message = "Running command"
+        self.print_progress()
+
+        self.start_time = time.time()
+
+        completed_proc = subprocess.run(args=formatted, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+
+        elapsed_time = time.time() - self.start_time
+
+        self.progress_message = "Parsing metrics"
+        self.print_progress()
+
+        ret_metrics = {}
+        if 'total_runtime' in in_metrics:
+            ret_metrics['total_runtime'] = { 'unit': 'seconds', 'value': elapsed_time }
+            in_metrics.remove('total_runtime')
+
+        # regex parse on output
+        if in_metrics:
```

Comment:
I think the driver should return metrics based on config file, not what was passed as part of the input. @pnickolov what do you think?